spec/rspec/core/world_spec.rb

```diff
@@ -21,6 +21,33 @@ module RSpec::Core
         world.register(example_group)
         expect(world.example_groups).to include(example_group)
       end
+
+      it "invokes before_register callbacks" do
+        world.before_register do |example_group|
+          examples     = example_group.examples
+          new_examples = []
+          examples.each do |ex|
+            caps = [{:browserName => :firefox}, {:browserName => :chrome}]
+            caps.each_with_index do |cap, index|
+              ex_with_cap = ex.clone
+              new_id      = "#{ex_with_cap.id}_#{index}_#{cap}"
+              ex_with_cap.instance_variable_set(:@id, new_id)
+              new_examples << ex_with_cap
+            end
+          end
+
+          example_group.instance_variable_set(:@examples, new_examples)
```

Comment:
This is how the feature is used (duplicate the existing examples in an example group and change their id/metadata). I can rewrite the spec to just modify the metadata which will avoid the instance variable calls.

I'm assuming you don't want users to have the ability to redefine examples on an example group?